packit/upstream.py

```diff
@@ -163,16 +163,33 @@ def create_patches(
 
         return patch_list
 
-    def get_upstream_version(self):
-        return (
-            versioneers_runner.run(
-                versioneer=None,
-                package_name=self.package_config.metadata["package_name"],
-                category=None,
-            )
-            or self.specfile.get_full_version()
+    def get_latest_released_version(self) -> str:
+        """
+        Return version of the upstream project for the latest official release
+
+        :return: the version string (e.g. "1.0.0")
+        """
+        version = versioneers_runner.run(
+            versioneer=None,
+            package_name=self.package_config.metadata["package_name"],
+            category=None,
         )
+        logger.info(f"Version in upstream registries is \"f{version}\".")
+        return version
 
+    def get_specfile_version(self) -> str:
+        """ provide version from specfile """
+        version = self.specfile.get_full_version()
+        logger.info(f"Version in spec file is \"f{version}\".")
+        return version
 
-class SourceGit(Upstream):
-    pass
+    def get_version(self) -> str:
+        """
+        Return version of latest release: prioritize upstream
+        package repositories over the version in spec
+        """
+        ups_ver = self.get_latest_released_version()
+        spec_ver = self.get_specfile_version()
+        if ups_ver:
+            return ups_ver
+        return spec_ver
```

Comment:
I would run `self.get_specfile_version()` after the if. It's not used in all cases.

Reply:
The only reason I am running both is that each function is logging the version from its respective source. I'll add a comment about it.

Reply:
done